.travis.yml

```diff
@@ -11,6 +10,10 @@ compiler:
     - gcc
     - clang
 
+env:
+    - {}
+    - FLAGS='-DUSE_PYTHON_3=ON'
```

Comment:
This would seem to either:
- add Python3 to all builds
- add a second set of builds that are Py3 but still required to pass for the PR to pass the gate.

I think the better solution right now would be to add a single build, perhaps Ubuntu 18.04 or 20.04, to check the Py3 functionality and allow it to fail.

Reply:
This expands the matrix so that each build (in the matrix only) either compiles regularly or with Python3.
result is:
- Bionic + gcc
- Bionic + clang
- Bionic + gcc + py3
- Bionic + clang + p3
- Xenial + gcc
- Xenial + clang

Reply:
![image](https://user-images.githubusercontent.com/1074110/86930131-a629ab00-c104-11ea-83f0-a23e6985c885.png)

Yes, we should have Xenial Py3 too at minimum; not sure why it only expanded the Bionic ones. We probably should add Focal (20.04) too.

Reply:
I tend to follow Ben on this

Reply:
Note: Focal isn't currently available, so ignore that part.

Reply:
So what's the bottom line for this PR? Add Xenial + Py3 and we're good to go?

Reply:
@nabaco sorry for the delay - xenial + py3 as an optional build that doesn't need to pass yet, just like we do for Mac.

Reply:
Done. Let's see the CI to make sure I haven't messed up anything